### PR TITLE
Fixed #62 and stopped generating new Fragment for every Transaction

### LINKED ISSUE #62: Hiding Bottom Navigation Bar when user scrolls to save space
**Expected Behaviour can be found here:**
https://storage.googleapis.com/spec-host-backup/mio-design%2Fassets%2F1i9Bgvs2vzCMkB3xtehQ37j8ktbCIZbI6%2Fbottomnav-scrolling-1b.mp4

### PR DESCRIPTION
**Pull request by aayushsingla**
Fixed #62 and stopped generating new Fragment for every Transaction

### CODE REVIEW

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/HomeActivity.java [hunk 1]
```diff
@@ -46,20 +51,41 @@
...
                 fragmentManager.beginTransaction().replace(R.id.container_fragment,
-                        CourseListFragment.newInstance()).commit();
+                        courseListFragment,"CourseListFragment").addToBackStack(null).commit();
                 return true;
             case R.id.navigation_profile:
+                fab_add_course.setVisibility(View.GONE);
+                fab_add_course.setOnClickListener(null);
+                Fragment profileFragment=fragmentManager.findFragmentByTag("ProfileFragment");
+                if(profileFragment==null){
+                    profileFragment=ProfileFragment.newInstance();
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/macbitsgoa/comrades/HomeActivity.java]: Delete these `.newInstance()` as they are plain constructor calls and nothing more.

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationFabBehavior.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.macbitsgoa.comrades
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import com.google.android.material.snackbar.Snackbar
+
+
+class BottomNavigationFabBehavior(context: Context?, attrs:AttributeSet?) : CoordinatorLayout.Behavior<View>(context,attrs) {
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationFabBehavior.kt]: where is this used?

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationViewBehavior.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import android.view.Gravity
+import android.view.View
+import androidx.annotation.RequiresApi
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.ViewCompat
+import com.google.android.material.snackbar.Snackbar
+import java.lang.Float.max
+import java.lang.Float.min
+
+class BottomNavigationViewBehavior<V : View>(context: Context, attrs: AttributeSet) :
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationViewBehavior.kt]: this is unused

Comment (aayushsingla) [reply androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationViewBehavior.kt]: it is used after in xml

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationFabBehavior.kt (hunk 2 above)
Comment (aayushsingla) [reply androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationFabBehavior.kt]: used in xml

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationViewBehavior.kt (hunk 3 above)
Comment (Rushi98) [reply androidClient/app/src/main/java/com/macbitsgoa/comrades/BottomNavigationViewBehavior.kt]: which xml?

**Pull request merged**